### PR TITLE
feat: Apply and Undeploy service use configured namespace

### PR DESCRIPTION
**Pull request by manusa**
## Description
feat: Apply and Undeploy service use configured Namespace

As opposed to using a property that was read from Maven and passed on
as a System property.

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have implemented unit tests to cover my changes
 - [ ] ~~I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly~~
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java [hunk 1]
```diff
@@ -140,6 +135,12 @@
     @Parameter(property = "jkube.environment")
     private String environment;
 
+    /**
+     * Namespace to use when accessing Kubernetes or OpenShift
+     */
+    @Parameter(property = "jkube.namespace")
+    protected String namespace;
```
Comment (rohanKanojia) [inline kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java:142]: Shall we move this to AbstractJKubeMojo?

Comment (manusa) [reply kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java:142]: I'm not sure that this applies to **all** mojos (e.g. build).
Currently, each Mojo has its own way of handling namespace.
So, I agree we should consider this, but not in the scope of this PR.

Comment (rohanKanojia) [reply kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java:142]: ohk

**Review by rohanKanojia — APPROVED** [review]

**Comment by manusa** [comment]
Relates to #570 #582

**Pull request merged**